### PR TITLE
CI: Update build flags for fedora 40

### PR DESCRIPTION
**Pull request by ikerexxe**
Now that fedora 40 is out let's update the flags.

### CODE REVIEW

File: share/containers/fedora.dockerfile [hunk 1]
```diff
@@ -15,16 +15,19 @@ RUN ./autogen.sh \
...
 	--with-sha-crypt \
 	--with-bcrypt \
 	--with-yescrypt \
 	--with-selinux \
-	--without-libpam \
-	--enable-shared \
 	--without-libbsd \
+	--without-libcrack \
+	--without-sssd \
+	--enable-shared \
```
Comment (alejandro-colomar) [inline share/containers/fedora.dockerfile:26]: Would you mind sorting `--enable-shared` in a separate commit from the one that changes flags?

Comment (alejandro-colomar) [reply share/containers/fedora.dockerfile:26]: Also, there are more `--enable-*` flags further down.  I guess if we're sorting it, that would be the best place, no?

File: share/containers/fedora.dockerfile [hunk 2]
```diff
@@ -15,16 +15,19 @@ RUN ./autogen.sh \
...
 	--without-libbsd \
-        --with-group-name-max-length=32 \
+	--without-libcrack \
+	--without-sssd \
+	--enable-shared \
+	--with-group-name-max-length=32 \
 	--enable-lastlog \
-	--enable-logind=no
+	--enable-logind=no \
+	--disable-account-tools-setuid
```
Comment (alejandro-colomar) [inline share/containers/fedora.dockerfile:30]: Oh, and there are also `--enable-*` ones up there.  Can we have them all together?  Or do you have reasons to prefer them in a different order?

Comment (alejandro-colomar) [reply share/containers/fedora.dockerfile:30]: I would use whatever order sort(1) produces.

File: share/containers/fedora.dockerfile (hunk 1 above)
Comment (alejandro-colomar) [reply share/containers/fedora.dockerfile:26]: This is what sort(1) produces:

```
	--disable-account-tools-setuid
	--enable-lastlog \
	--enable-logind=no \
	--enable-man \
	--enable-shadowgrp \
	--enable-shared \
	--with-audit \
	--with-bcrypt \
	--with-group-name-max-length=32 \
	--with-libpam \
	--with-selinux \
	--with-sha-crypt \
	--with-yescrypt \
	--without-libbsd \
	--without-libcrack \
	--without-sssd \
```

Comment (ikerexxe) [reply share/containers/fedora.dockerfile]: Updated

File: share/containers/fedora.dockerfile (hunk 2 above)
Comment (ikerexxe) [reply share/containers/fedora.dockerfile]: Updated

File: share/containers/fedora.dockerfile [hunk 3]
```diff
@@ -12,19 +12,19 @@ COPY ./ /usr/local/src/shadow/
...
 	--with-selinux \
-	--without-libpam \
-	--enable-shared \
+	--with-sha-crypt \
+	--with-yescrypt \
 	--without-libbsd \
-	--with-group-name-max-length=32 \
-	--enable-lastlog \
-	--enable-logind=no
+	--without-libpam
```
Comment (alejandro-colomar) [inline share/containers/fedora.dockerfile]: The sort LGTM.

```sh
$ diff -u <(grep ^+ diff | sed s/.// | sort) <(grep ^- diff | sed s/.// | sort) 
--- /dev/fd/63	2024-04-29 11:36:44.457595901 +0200
+++ /dev/fd/62	2024-04-29 11:36:44.457595901 +0200
@@ -1,8 +1,8 @@
 	--enable-lastlog \
-	--enable-logind=no \
+	--enable-logind=no
 	--enable-shadowgrp \
 	--enable-shared \
 	--with-group-name-max-length=32 \
 	--with-sha-crypt \
 	--with-yescrypt \
-	--without-libpam
+	--without-libpam \
```

**Review by alejandro-colomar — COMMENTED** [review]
The first two commits LGTM.  For the last, maybe it would be good to document in the commit message why those need to change.  I assume it's just due to changes in Fedora 40, and I trust you know what you're doing.  But especially after Recent Events, I guess it's better to state it explicitly.

For the first two commits (the refactors):

```
Reviewed-by: Alejandro Colomar <alx@kernel.org>
```

**Comment by ikerexxe** [comment]
Updated to tackle your last comments.

**Review by alejandro-colomar — APPROVED** [review]
Thanks!

**Pull request merged**